src/inmanta/protocol/rest/server.py

```diff
@@ -44,6 +44,12 @@ def initialize(self, transport: "RESTServer", config: Dict[str, common.UrlMethod
         self._transport: "RESTServer" = transport
         self._config = config
 
+        # Setting "Access-Control-Allow-Origin": null can be exploited.
+        # better not set it all instead.
+        # See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin
+        if inmanta_config.Config.get("server", "access_control_allow_origin"):
+            self.set_header("Access-Control-Allow-Origin", inmanta_config.Config.get("server", "access_control_allow_origin"))
```

Comment:
Are you allowed to set headers already in the initialize method? If I look at the tornado docs it seems that this should go in the prepare method and not in the initialize. Initialize is called at the end of the constructor of the baseclass. prepare is only called after the first sanity check of the request have passed.

Reply:
This is a good point, I looked further in to it.
We should implemend `RequestHandler.set_default_headers` instead and move the header init to there instead:
https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.set_default_headers